overlord/snapstate/autorefresh_gating.go

```diff
@@ -205,6 +252,9 @@ func HoldRefresh(st *state.State, gatingSnap string, holdDuration time.Duration,
 
 		newHold := now.Add(dur)
 		cutOff := lastRefreshTime.Add(maxPostponement - maxPostponementBuffer)
+		if gatingSnap == "system" {
+			cutOff = now.Add(maxDuration)
+		}
 
```

Comment:
I wonder whether instead of the single mutltiple ifs about "system" here and in maxAllowedPostment there should't be a more general if system else  in this function? if the worry is too much identation, we could split out a computeCutOff helper?

Reply:
I split out a more general case for "system" like you suggested. I think it's simpler because it doesn't complicate the existing branch and since we don't restrict the max duration/postponement, we can just set `holdUntil` to anything